### DISCUSSION TITLE
Update a Ike.Gateway

### DISCUSSION THREAD

**Discussion by jvictorgonc** [PaloAltoNetworks/pan-os-python#361] [Q&A]
HI All,
i want to enable IKEv2 in some IKE Gateways, however the command i can´t.

When i try to use the "update()" after the object instanced, i receive the error "TypeError: update() missing 1 required positional argument: 'variable'".

Follow the commands:

fw = firewall.Firewall('192.168.1.1', api_username=api_user, api_password=api_password)

ikegw = network.IkeGateway(name=line[0], peer_ip_value=line[1], pre_shared_key=line[-1], interface='ae1.111', ikev1_crypto_profile=line[3], version='ikev2')

fw.add(ikegw).update()

Someone can help me?

I´m a beginer in python and pandevice...

Thanks,
JVictor

**Comment by shinmog** [comment]
Update is to change a single parameter / variable of an object on the live PAN-OS, which requires that you give the parameter / variable name you want to update.  The upside to using this is the XML sent is much smaller / concise.

https://pan-os-python.readthedocs.io/en/latest/module-base.html#panos.base.PanObject.update

If you want to change multiple params, you can use `.apply()`.

If you only need to change one thing, you can use either method.
